core/trino-main/src/main/java/io/trino/sql/planner/plan/AggregationNode.java

```diff
@@ -100,12 +108,34 @@ public AggregationNode(
         checkArgument(preGroupedSymbols.isEmpty() || groupingSets.getGroupingKeys().containsAll(preGroupedSymbols), "Pre-grouped symbols must be a subset of the grouping keys");
         this.preGroupedSymbols = ImmutableList.copyOf(preGroupedSymbols);
 
-        ImmutableList.Builder<Symbol> outputs = ImmutableList.builder();
-        outputs.addAll(groupingSets.getGroupingKeys());
-        hashSymbol.ifPresent(outputs::add);
-        outputs.addAll(aggregations.keySet());
+        ImmutableList.Builder<Symbol> outputsBuilder = ImmutableList.builder();
+        outputsBuilder.addAll(groupingSets.getGroupingKeys());
+        hashSymbol.ifPresent(outputsBuilder::add);
+        outputsBuilder.addAll(aggregations.keySet());
+        if (step.isOutputPartial() && rawInputMaskSymbol.isPresent()) {
```

Comment:
why we cannot end-up with PA and distinct or ordered aggregation?

Will PA (with `mask`) be added when `optimizer.use-mark-distinct=false`?

Reply:
> why we cannot end-up with PA and distinct or ordered aggregation?

see `io.trino.sql.planner.iterative.rule.PushPartialAggregationThroughExchange#apply`

```
boolean decomposable = aggregationNode.isDecomposable(context.getSession(), plannerContext.getMetadata());
...
        if (!decomposable || !preferPartialAggregation(context.getSession())) {
            return Result.empty();
        }
```

> Will PA (with mask) be added when optimizer.use-mark-distinct=false?

Mask is also used by `ImplementFilteredAggregations`.
btw I don't think it matters, it is used in the aggregators so we have to push it through is it's in the `AggregationNode`.